### ISSUE TITLE
WGSL is missing a `sample_mask` builtin.

### ISSUE THREAD

**Issue by Kangz** [gpuweb/gpuweb#1259]
All APIs we target have a builtin to access the sample mask of the current fragment invocation, both for reading (what samples does the fragment cover) and writing (which samples the fragment should cover). We should add it to WGSL.

I would have done it myself, but this builtin has the particularity that it is both an input and an output of the fragment stage. So it's not clear what should happen with it, is it `var<inout>` or do we separate it into two different builtins `sample_mask_in` and `sample_mask_out`?

**Comment by kvark** [comment]
> what samples does the fragment cover

This could be a bit tricky. An investigation is still needed. One thing to clarify between the APIs is whether or not the early depth/stencil test affects the sample mask as read by the shader.

> is it var<inout> or do we separate it into two different builtins

It could be just 2 different vars:
```
[[builtin(sample_mask)]]
var<in> sampleMaskIn;
[[builtin(sample_mask)]]
var<out> sampleMaskOut;
```

**Comment by Kangz** [comment]
> An investigation is still needed. One thing to clarify between the APIs is whether or not the early depth/stencil test affects the sample mask as read by the shader.

Agreed that we need the clarification eventually, but it's pretty clear that whatever the semantic of the input sample mask, we need to have the builtin in WGSL.

**Comment by grorg** [comment]
Discussed at the [2020-12-01 meeting](https://docs.google.com/document/d/1BGljZlq1ZY1DTZS_MHxTxfu6rF0j87uUrPL7RHArpLA/).

**Comment by grorg** [comment]
Resolved: Yes, we will have this. There will be an investigation into existing backends to help guide naming, but will likely end up with two variables (one for in, one for out).

**Comment by kvark** [comment]
Fixed by #1318
